tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -255,12 +265,17 @@ public static Trace getAndClearTrace() {
      * Tracer#completeSpan span completion}.
      */
     public static boolean isTraceObservable() {
-        return currentTrace.get().isObservable();
+        Trace trace = currentTrace.get();
+        return trace != null && trace.isObservable();
```

Comment:
related to above comments, could add `isObservable(Trace)` to avoid additional thread local lookups

```suggestion        
        return isObservable(currentTrace.get());
    }

    private static boolean isObservable(Trace trace) {
        return trace != null && trace.isObservable();
    }
```